mono/mini/exceptions-arm64.c

```diff
@@ -620,3 +626,598 @@ mono_arch_do_ip_adjustment (MonoContext *ctx)
+	unwind_codes[(*unwind_code_size)++] = 0b11100100;
+}
+
+static void
+mono_arch_unwind_add_end_c (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100101;
+}
+
+static void
+mono_arch_unwind_add_save_fplr (guint8* unwind_codes, guint32* unwind_code_size, gint32 offset) {
+
+	mono_arch_unwind_add_assert (1, 504, *unwind_code_size, offset);
+
+	unwind_codes[(*unwind_code_size)++] = 0b01000000 | offset / 8;
+}
+
+static void
+mono_arch_unwind_add_save_fplr_x (guint8* unwind_codes, guint32* unwind_code_size, gint32 offset) {
+
+	mono_arch_unwind_add_assert (1, 512, *unwind_code_size, offset);
+
+	unwind_codes[(*unwind_code_size)++] = 0b10000000 | (offset-8) / 8;
+}
+
+static void
+mono_arch_unwind_add_save_reg (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 504, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11010000 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
+}
+
+static void
+mono_arch_unwind_add_save_reg_x (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 256, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11010000 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
+}
+
+static void
+mono_arch_unwind_add_save_regp (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 504, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11001000 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
+}
+
+static void
+mono_arch_unwind_add_save_regp_x (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 256, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11001100 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
+}
+
+static void
+mono_arch_unwind_add_alloc_s (guint8* unwind_codes, guint32* unwind_code_size, guint32 size) {
+
+	mono_arch_unwind_add_assert (1, 511, *unwind_code_size, size);
+	unwind_codes[(*unwind_code_size)++] = 0b00000000 | (size / MONO_ARCH_FRAME_ALIGNMENT);
```

Comment:
Assert that this is aligned?

Reply:
Added in https://github.com/Unity-Technologies/mono/pull/1960/commits/9b8e291ee6dde9c51871bf05de6ad195e1d48f01